core/db_models/EEM_Ticket.model.php

```diff
@@ -383,10 +379,73 @@ public function get_tickets_with_IDs(array $ticket_IDs)
     }
 
 
+    /**
+     * @return void
+     */
+    private function parseTicketVisibilityOptions()
+    {
+        $this->ticket_visibility = (array) apply_filters(
+            'FHEE__EEM_Ticket__construct__ticket_visibility',
+            [
+                EEM_Ticket::TICKET_VISIBILITY_PUBLIC_KEY        => [
+                    'label' => esc_html__('Public', 'event_espresso'),
+                    'value' => EEM_Ticket::TICKET_VISIBILITY_PUBLIC_VALUE,
+                ],
+                EEM_Ticket::TICKET_VISIBILITY_MEMBERS_ONLY_KEY  => [
+                    'label' => esc_html__('Members only', 'event_espresso'),
+                    'value' => EEM_Ticket::TICKET_VISIBILITY_MEMBERS_ONLY_VALUE,
+                ],
+                EEM_Ticket::TICKET_VISIBILITY_ADMINS_ONLY_KEY   => [
+                    'label' => esc_html__('Admins only', 'event_espresso'),
+                    'value' => EEM_Ticket::TICKET_VISIBILITY_ADMINS_ONLY_VALUE,
+                ],
+                EEM_Ticket::TICKET_VISIBILITY_ADMIN_UI_ONLY_KEY => [
+                    'label' => esc_html__('Admin UI only', 'event_espresso'),
+                    'value' => EEM_Ticket::TICKET_VISIBILITY_ADMIN_UI_ONLY_VALUE,
+                ],
+                EEM_Ticket::TICKET_VISIBILITY_NONE_KEY          => [
+                    'label' => esc_html__('None', 'event_espresso'),
+                    'value' => EEM_Ticket::TICKET_VISIBILITY_NONE_VALUE,
+                ],
+            ]
+        );
+    }
+
+
+    /**
+     * @return array
+     */
+    public function getTicketVisibilityEnumOptions(): array
+    {
+        $ticket_visibility = [];
+        foreach ($this->ticket_visibility as $visibility) {
+            if (isset($visibility['value'], $visibility['label'])) {
+                $ticket_visibility[ $visibility['value'] ] = $visibility['label'];
+            }
+        }
+        return $ticket_visibility;
+    }
+
+
+    /**
+     * @return array
+     */
+    public function getTicketVisibilityKeyValuePairs(): array
+    {
+        $ticket_visibility = [];
+        foreach ($this->ticket_visibility as $key => $ticket_visibility_option) {
+            if (isset($ticket_visibility_option['value'])) {
+                $ticket_visibility[ $key ] = $ticket_visibility_option['value'];
```

Comment:
```suggestion
                $name = str_replace('TICKET_VISIBILITY_', '', $key);
                $ticket_visibility[ $name ] = ['value' => $ticket_visibility_option['value']];
```

Reply:
i added new constants for those keys, like:

```php
public const TICKET_VISIBILITY_PUBLIC_KEY          = 'PUBLIC';
```